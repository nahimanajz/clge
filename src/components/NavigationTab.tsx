"use client";
import { INavbarProps } from "@/shared/props";
import React, { FC, useEffect, useState } from "react";
import { FiSunrise } from "react-icons/fi";

const NavigationTab: FC<INavbarProps> = ({ onFilter, isTyping }) => {
  const [days, setDays] = useState([
    {
      day: "Monday",
      isActive: false,
    },
    {
      day: "Tuesday",
      isActive: false,
    },
    {
      day: "Wednesday",
      isActive: false,
    },
    {
      day: "Thursday",
      isActive: false,
    },
    {
      day: "Friday",
      isActive: false,
    },
    {
      day: "Saturday",
      isActive: false,
    },
    {
      day: "Sunday",
      isActive: false,
    },
  ]);

  const [menuOpen, setMenuOpen] = useState(false);

  const toggleMenu = () => {
    setMenuOpen(!menuOpen);
  };

  const handleDayClick = (clickedDay: string) => {
    const updatedDays = days.map((day) => ({
      ...day,
      isActive: day.day === clickedDay ? true : false,
    }));
    setDays(updatedDays);
  };
  useEffect(()=> {
if(isTyping){
  setDays(days.map((day)=> ({...day, isActive: false})))
}
  }, [isTyping])
  return (
    <div className="p-4">
      <div className="flex justify-between"> 
        <div className="hidden md:flex space-x-4 justify-center transition delay-300 duration-300 ease-in-out hover:transform hover:scale-105 w-full"> 
          {days.map((day) => (
            <div
              className="flex flex-col justify-center self-center cursor-pointer"
              key={day.day}
              onClick={() => {
                onFilter(day.day, false);
                handleDayClick(day.day);
              }}
            >
              <FiSunrise className="w-5 h-5 self-center" />
              <div className={`text-gray font-semibold hover:text-dark my-3 `}>
                {day.day}
              </div>
              {day.isActive ? <hr className="text-dark"/> : ""}
            </div>
          ))}
        </div>
        <div className="flex md:hidden items-center">
          <button onClick={toggleMenu}>
            <svg
              xmlns="http://www.w3.org/2000/svg"
              className="h-6 w-6 cursor-pointer text-gray"
              fill="none"
              viewBox="0 0 24 24"
              stroke="currentColor"
            >
              {menuOpen ? (
                <path
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth={2}
                  d="M6 18L18 6M6 6l12 12"
                />
              ) : (
                <path
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth={2}
                  d="M4 6h16M4 12h16M4 18h16"
                />
              )}
            </svg>
          </button>
        </div>
      </div>
      {menuOpen && (
        <div className="md:hidden flex flex-col mt-4 space-y-2">
          {days.map((day) => (
            <div
              className="flex flex-row cursor-pointer text-gray font-semibold hover:text-dark"
              key={day.day}
              onClick={() => {
                onFilter(day.day, false);
                handleDayClick(day.day);
              }}
            >
              <FiSunrise className="w-5 h-5 mr-4" />
              <div
                className={` ${
                  day.isActive ? "border-b-[3px] border-primary" : ""
                }`}
              >
                {day.day}
              </div>
            </div>
          ))}
        </div>
      )}
    </div>
  );

};

export default NavigationTab;
