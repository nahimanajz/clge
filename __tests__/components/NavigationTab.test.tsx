import NavigationTab from "@/components/NavigationTab";
import { render, screen, fireEvent } from "@testing-library/react";


const onClick = jest.fn()

describe("Navigation ", () => {
  it(" should match snapshot", () => {
    const onFilter = () => {};
    const { container } = render(<NavigationTab onFilter={onFilter} />);
    expect(container).toMatchSnapshot();
  });

  it("should handle click event", ()=> {
    const onFilter = () => {};
    const { container } = render(<NavigationTab onFilter={onFilter} />);
    expect(container).toMatchSnapshot();
  })
});
