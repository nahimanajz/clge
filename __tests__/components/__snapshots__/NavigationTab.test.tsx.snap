// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation   should match snapshot 1`] = `
<div>
  <div
    class="p-4"
  >
    <div
      class="flex justify-between"
    >
      <div
        class="hidden md:flex space-x-4 justify-center transition delay-300 duration-300 ease-in-out hover:transform hover:scale-105 w-full"
      >
        <div
          class="flex flex-col justify-center self-center cursor-pointer"
        >
          <svg
            class="w-5 h-5 self-center"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17 18a5 5 0 0 0-10 0"
            />
            <line
              x1="12"
              x2="12"
              y1="2"
              y2="9"
            />
            <line
              x1="4.22"
              x2="5.64"
              y1="10.22"
              y2="11.64"
            />
            <line
              x1="1"
              x2="3"
              y1="18"
              y2="18"
            />
            <line
              x1="21"
              x2="23"
              y1="18"
              y2="18"
            />
            <line
              x1="18.36"
              x2="19.78"
              y1="11.64"
              y2="10.22"
            />
            <line
              x1="23"
              x2="1"
              y1="22"
              y2="22"
            />
            <polyline
              points="8 6 12 2 16 6"
            />
          </svg>
          <div
            class="text-gray font-semibold hover:text-dark my-3 "
          >
            Monday
          </div>
        </div>
        <div
          class="flex flex-col justify-center self-center cursor-pointer"
        >
          <svg
            class="w-5 h-5 self-center"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17 18a5 5 0 0 0-10 0"
            />
            <line
              x1="12"
              x2="12"
              y1="2"
              y2="9"
            />
            <line
              x1="4.22"
              x2="5.64"
              y1="10.22"
              y2="11.64"
            />
            <line
              x1="1"
              x2="3"
              y1="18"
              y2="18"
            />
            <line
              x1="21"
              x2="23"
              y1="18"
              y2="18"
            />
            <line
              x1="18.36"
              x2="19.78"
              y1="11.64"
              y2="10.22"
            />
            <line
              x1="23"
              x2="1"
              y1="22"
              y2="22"
            />
            <polyline
              points="8 6 12 2 16 6"
            />
          </svg>
          <div
            class="text-gray font-semibold hover:text-dark my-3 "
          >
            Tuesday
          </div>
        </div>
        <div
          class="flex flex-col justify-center self-center cursor-pointer"
        >
          <svg
            class="w-5 h-5 self-center"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17 18a5 5 0 0 0-10 0"
            />
            <line
              x1="12"
              x2="12"
              y1="2"
              y2="9"
            />
            <line
              x1="4.22"
              x2="5.64"
              y1="10.22"
              y2="11.64"
            />
            <line
              x1="1"
              x2="3"
              y1="18"
              y2="18"
            />
            <line
              x1="21"
              x2="23"
              y1="18"
              y2="18"
            />
            <line
              x1="18.36"
              x2="19.78"
              y1="11.64"
              y2="10.22"
            />
            <line
              x1="23"
              x2="1"
              y1="22"
              y2="22"
            />
            <polyline
              points="8 6 12 2 16 6"
            />
          </svg>
          <div
            class="text-gray font-semibold hover:text-dark my-3 "
          >
            Wednesday
          </div>
        </div>
        <div
          class="flex flex-col justify-center self-center cursor-pointer"
        >
          <svg
            class="w-5 h-5 self-center"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17 18a5 5 0 0 0-10 0"
            />
            <line
              x1="12"
              x2="12"
              y1="2"
              y2="9"
            />
            <line
              x1="4.22"
              x2="5.64"
              y1="10.22"
              y2="11.64"
            />
            <line
              x1="1"
              x2="3"
              y1="18"
              y2="18"
            />
            <line
              x1="21"
              x2="23"
              y1="18"
              y2="18"
            />
            <line
              x1="18.36"
              x2="19.78"
              y1="11.64"
              y2="10.22"
            />
            <line
              x1="23"
              x2="1"
              y1="22"
              y2="22"
            />
            <polyline
              points="8 6 12 2 16 6"
            />
          </svg>
          <div
            class="text-gray font-semibold hover:text-dark my-3 "
          >
            Thursday
          </div>
        </div>
        <div
          class="flex flex-col justify-center self-center cursor-pointer"
        >
          <svg
            class="w-5 h-5 self-center"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17 18a5 5 0 0 0-10 0"
            />
            <line
              x1="12"
              x2="12"
              y1="2"
              y2="9"
            />
            <line
              x1="4.22"
              x2="5.64"
              y1="10.22"
              y2="11.64"
            />
            <line
              x1="1"
              x2="3"
              y1="18"
              y2="18"
            />
            <line
              x1="21"
              x2="23"
              y1="18"
              y2="18"
            />
            <line
              x1="18.36"
              x2="19.78"
              y1="11.64"
              y2="10.22"
            />
            <line
              x1="23"
              x2="1"
              y1="22"
              y2="22"
            />
            <polyline
              points="8 6 12 2 16 6"
            />
          </svg>
          <div
            class="text-gray font-semibold hover:text-dark my-3 "
          >
            Friday
          </div>
        </div>
        <div
          class="flex flex-col justify-center self-center cursor-pointer"
        >
          <svg
            class="w-5 h-5 self-center"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17 18a5 5 0 0 0-10 0"
            />
            <line
              x1="12"
              x2="12"
              y1="2"
              y2="9"
            />
            <line
              x1="4.22"
              x2="5.64"
              y1="10.22"
              y2="11.64"
            />
            <line
              x1="1"
              x2="3"
              y1="18"
              y2="18"
            />
            <line
              x1="21"
              x2="23"
              y1="18"
              y2="18"
            />
            <line
              x1="18.36"
              x2="19.78"
              y1="11.64"
              y2="10.22"
            />
            <line
              x1="23"
              x2="1"
              y1="22"
              y2="22"
            />
            <polyline
              points="8 6 12 2 16 6"
            />
          </svg>
          <div
            class="text-gray font-semibold hover:text-dark my-3 "
          >
            Saturday
          </div>
        </div>
        <div
          class="flex flex-col justify-center self-center cursor-pointer"
        >
          <svg
            class="w-5 h-5 self-center"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17 18a5 5 0 0 0-10 0"
            />
            <line
              x1="12"
              x2="12"
              y1="2"
              y2="9"
            />
            <line
              x1="4.22"
              x2="5.64"
              y1="10.22"
              y2="11.64"
            />
            <line
              x1="1"
              x2="3"
              y1="18"
              y2="18"
            />
            <line
              x1="21"
              x2="23"
              y1="18"
              y2="18"
            />
            <line
              x1="18.36"
              x2="19.78"
              y1="11.64"
              y2="10.22"
            />
            <line
              x1="23"
              x2="1"
              y1="22"
              y2="22"
            />
            <polyline
              points="8 6 12 2 16 6"
            />
          </svg>
          <div
            class="text-gray font-semibold hover:text-dark my-3 "
          >
            Sunday
          </div>
        </div>
      </div>
      <div
        class="flex md:hidden items-center"
      >
        <button>
          <svg
            class="h-6 w-6 cursor-pointer text-gray"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M4 6h16M4 12h16M4 18h16"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Navigation  should handle click event 1`] = `
<div>
  <div
    class="p-4"
  >
    <div
      class="flex justify-between"
    >
      <div
        class="hidden md:flex space-x-4 justify-center transition delay-300 duration-300 ease-in-out hover:transform hover:scale-105 w-full"
      >
        <div
          class="flex flex-col justify-center self-center cursor-pointer"
        >
          <svg
            class="w-5 h-5 self-center"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17 18a5 5 0 0 0-10 0"
            />
            <line
              x1="12"
              x2="12"
              y1="2"
              y2="9"
            />
            <line
              x1="4.22"
              x2="5.64"
              y1="10.22"
              y2="11.64"
            />
            <line
              x1="1"
              x2="3"
              y1="18"
              y2="18"
            />
            <line
              x1="21"
              x2="23"
              y1="18"
              y2="18"
            />
            <line
              x1="18.36"
              x2="19.78"
              y1="11.64"
              y2="10.22"
            />
            <line
              x1="23"
              x2="1"
              y1="22"
              y2="22"
            />
            <polyline
              points="8 6 12 2 16 6"
            />
          </svg>
          <div
            class="text-gray font-semibold hover:text-dark my-3 "
          >
            Monday
          </div>
        </div>
        <div
          class="flex flex-col justify-center self-center cursor-pointer"
        >
          <svg
            class="w-5 h-5 self-center"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17 18a5 5 0 0 0-10 0"
            />
            <line
              x1="12"
              x2="12"
              y1="2"
              y2="9"
            />
            <line
              x1="4.22"
              x2="5.64"
              y1="10.22"
              y2="11.64"
            />
            <line
              x1="1"
              x2="3"
              y1="18"
              y2="18"
            />
            <line
              x1="21"
              x2="23"
              y1="18"
              y2="18"
            />
            <line
              x1="18.36"
              x2="19.78"
              y1="11.64"
              y2="10.22"
            />
            <line
              x1="23"
              x2="1"
              y1="22"
              y2="22"
            />
            <polyline
              points="8 6 12 2 16 6"
            />
          </svg>
          <div
            class="text-gray font-semibold hover:text-dark my-3 "
          >
            Tuesday
          </div>
        </div>
        <div
          class="flex flex-col justify-center self-center cursor-pointer"
        >
          <svg
            class="w-5 h-5 self-center"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17 18a5 5 0 0 0-10 0"
            />
            <line
              x1="12"
              x2="12"
              y1="2"
              y2="9"
            />
            <line
              x1="4.22"
              x2="5.64"
              y1="10.22"
              y2="11.64"
            />
            <line
              x1="1"
              x2="3"
              y1="18"
              y2="18"
            />
            <line
              x1="21"
              x2="23"
              y1="18"
              y2="18"
            />
            <line
              x1="18.36"
              x2="19.78"
              y1="11.64"
              y2="10.22"
            />
            <line
              x1="23"
              x2="1"
              y1="22"
              y2="22"
            />
            <polyline
              points="8 6 12 2 16 6"
            />
          </svg>
          <div
            class="text-gray font-semibold hover:text-dark my-3 "
          >
            Wednesday
          </div>
        </div>
        <div
          class="flex flex-col justify-center self-center cursor-pointer"
        >
          <svg
            class="w-5 h-5 self-center"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17 18a5 5 0 0 0-10 0"
            />
            <line
              x1="12"
              x2="12"
              y1="2"
              y2="9"
            />
            <line
              x1="4.22"
              x2="5.64"
              y1="10.22"
              y2="11.64"
            />
            <line
              x1="1"
              x2="3"
              y1="18"
              y2="18"
            />
            <line
              x1="21"
              x2="23"
              y1="18"
              y2="18"
            />
            <line
              x1="18.36"
              x2="19.78"
              y1="11.64"
              y2="10.22"
            />
            <line
              x1="23"
              x2="1"
              y1="22"
              y2="22"
            />
            <polyline
              points="8 6 12 2 16 6"
            />
          </svg>
          <div
            class="text-gray font-semibold hover:text-dark my-3 "
          >
            Thursday
          </div>
        </div>
        <div
          class="flex flex-col justify-center self-center cursor-pointer"
        >
          <svg
            class="w-5 h-5 self-center"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17 18a5 5 0 0 0-10 0"
            />
            <line
              x1="12"
              x2="12"
              y1="2"
              y2="9"
            />
            <line
              x1="4.22"
              x2="5.64"
              y1="10.22"
              y2="11.64"
            />
            <line
              x1="1"
              x2="3"
              y1="18"
              y2="18"
            />
            <line
              x1="21"
              x2="23"
              y1="18"
              y2="18"
            />
            <line
              x1="18.36"
              x2="19.78"
              y1="11.64"
              y2="10.22"
            />
            <line
              x1="23"
              x2="1"
              y1="22"
              y2="22"
            />
            <polyline
              points="8 6 12 2 16 6"
            />
          </svg>
          <div
            class="text-gray font-semibold hover:text-dark my-3 "
          >
            Friday
          </div>
        </div>
        <div
          class="flex flex-col justify-center self-center cursor-pointer"
        >
          <svg
            class="w-5 h-5 self-center"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17 18a5 5 0 0 0-10 0"
            />
            <line
              x1="12"
              x2="12"
              y1="2"
              y2="9"
            />
            <line
              x1="4.22"
              x2="5.64"
              y1="10.22"
              y2="11.64"
            />
            <line
              x1="1"
              x2="3"
              y1="18"
              y2="18"
            />
            <line
              x1="21"
              x2="23"
              y1="18"
              y2="18"
            />
            <line
              x1="18.36"
              x2="19.78"
              y1="11.64"
              y2="10.22"
            />
            <line
              x1="23"
              x2="1"
              y1="22"
              y2="22"
            />
            <polyline
              points="8 6 12 2 16 6"
            />
          </svg>
          <div
            class="text-gray font-semibold hover:text-dark my-3 "
          >
            Saturday
          </div>
        </div>
        <div
          class="flex flex-col justify-center self-center cursor-pointer"
        >
          <svg
            class="w-5 h-5 self-center"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17 18a5 5 0 0 0-10 0"
            />
            <line
              x1="12"
              x2="12"
              y1="2"
              y2="9"
            />
            <line
              x1="4.22"
              x2="5.64"
              y1="10.22"
              y2="11.64"
            />
            <line
              x1="1"
              x2="3"
              y1="18"
              y2="18"
            />
            <line
              x1="21"
              x2="23"
              y1="18"
              y2="18"
            />
            <line
              x1="18.36"
              x2="19.78"
              y1="11.64"
              y2="10.22"
            />
            <line
              x1="23"
              x2="1"
              y1="22"
              y2="22"
            />
            <polyline
              points="8 6 12 2 16 6"
            />
          </svg>
          <div
            class="text-gray font-semibold hover:text-dark my-3 "
          >
            Sunday
          </div>
        </div>
      </div>
      <div
        class="flex md:hidden items-center"
      >
        <button>
          <svg
            class="h-6 w-6 cursor-pointer text-gray"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M4 6h16M4 12h16M4 18h16"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;
